Tests/JWEHeaderTests.swift

```diff
@@ -125,4 +125,61 @@ class JWEHeaderTests: XCTestCase {
         XCTFail()
     }
 
+    func testSetHeaderParameters() {
+        let jku: URL? = nil
+        let jwk = "jwk"
+        let kid = "kid"
+        let x5u: URL? = nil
+        let x5c: [String: Any]? = ["":""]
+        let x5t = "x5t"
+        let x5tS256 = "x5tS256"
+        let typ = "typ"
+        let cty = "cty"
+        let crit = ["crit1", "crit2"]
+
+        var header = JWEHeader(algorithm: .RSA1_5, encryptionAlgorithm: .A256CBCHS512)
+        header.jku = jku
+        header.jwk = jwk
+        header.kid = kid
+        header.x5u = x5u
+        header.x5c = x5c
+        header.x5t = x5t
+        header.x5tS256 = x5tS256
+        header.typ = typ
+        header.cty = cty
+        header.crit = crit
+
+        XCTAssertEqual(header.data(), try! JSONSerialization.data(withJSONObject: header.parameters, options: []))
+
+        XCTAssertEqual(header.parameters["jku"] as? URL, jku)
+        XCTAssertEqual(header.jku, jku)
+
+        XCTAssertEqual(header.parameters["jwk"] as? String, jwk)
+        XCTAssertEqual(header.jwk, jwk)
+
+        XCTAssertEqual(header.parameters["kid"] as? String, kid)
+        XCTAssertEqual(header.kid, kid)
+
+        XCTAssertEqual(header.parameters["x5u"] as? URL, x5u)
+        XCTAssertEqual(header.x5u, x5u)
+
+//        XCTAssertEqual(header.parameters["x5c"] as? [String: Any], x5c)
```

Comment:
I don't get these two commented lines to work, maybe you can have a look at them?

Reply:
This is probably because Dictionaries are not conforming to the `Equatable` protocol.

Reply:
As noted above, you can change the type of `x5c` to `[String]`. This should solve this problem.

Reply:
@daniel-mohemian Perfect, with `[String]` this is no problem anymore ;)